crypto/src/mls/conversation/conversation_guard/encrypt.rs

```diff
@@ -0,0 +1,126 @@
+//! The methods in this module are concerned with message encryption.
+
+use super::ConversationGuard;
+use super::Result;
+use crate::mls::conversation::Error;
+use crate::MlsError;
+use openmls::prelude::MlsMessageOutBody;
+
+impl ConversationGuard {
+    /// Encrypts a raw payload then serializes it to the TLS wire format
+    /// Can only be called when there is no pending commit and no pending proposal.
+    ///
+    /// # Arguments
+    /// * `message` - the message as a byte array
+    ///
+    /// # Return type
+    /// This method will return an encrypted TLS serialized message.
+    ///
+    /// # Errors
+    /// If the conversation can't be found, an error will be returned. Other errors are originating
+    /// from OpenMls and the KeyStore
+    pub async fn encrypt_message(&mut self, message: impl AsRef<[u8]>) -> Result<Vec<u8>> {
+        let client = self.mls_client().await?;
+        let backend = self.mls_provider().await?;
+        let mut inner = self.conversation_mut().await;
+        let signer = &inner
+            .find_current_credential_bundle(&client)
+            .await
+            .map_err(|_| Error::IdentityInitializationError)?
+            .signature_key;
```

Comment:
We've used this pattern in a few places already. I wonder if it's worth extracting it to a helper method like we did for `mls_client` and `mls_provider`.